container-reverse-proxy.tf

```diff
@@ -0,0 +1,368 @@
+  security_group_id        = data.terraform_remote_state.ingest.outputs.emr_common_sg.id
+  provider                 = aws.target
+}
+
+resource "aws_security_group_rule" "egress_rm_endpoint" {
+  depends_on               = [aws_vpc_peering_connection_accepter.reverse_proxy_ingest[0]]
+  count                    = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  description              = "Allow nginx reverse proxy container to reach Yarn ResourceManager UI"
+  type                     = "egress"
+  from_port                = 8088
+  to_port                  = 8088
+  protocol                 = "tcp"
+  source_security_group_id = data.terraform_remote_state.ingest.outputs.emr_common_sg.id
+  security_group_id        = aws_security_group.reverse_proxy_ecs[0].id
+}
+
+resource "aws_security_group_rule" "ingress_rm_endpoint" {
+  depends_on               = [aws_vpc_peering_connection_accepter.reverse_proxy_ingest[0]]
+  count                    = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  description              = "Allow Yarn ResourceManager UI to be reached by nginx reverse proxy container"
+  type                     = "ingress"
+  from_port                = 8088
+  to_port                  = 8088
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.reverse_proxy_ecs[0].id
+  security_group_id        = data.terraform_remote_state.ingest.outputs.emr_common_sg.id
+  provider                 = aws.target
+}
+
+resource "aws_security_group_rule" "egress_internet_proxy" {
+  count                    = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  description              = "Allow Internet access via the proxy for reverse proxy container"
+  type                     = "egress"
+  from_port                = 3128
+  to_port                  = 3128
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.internet_proxy_endpoint.id
+  security_group_id        = aws_security_group.reverse_proxy_ecs[0].id
+}
+
+resource "aws_security_group_rule" "ingress_internet_proxy" {
+  count                    = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  description              = "Allow proxy access from reverse proxy container"
+  type                     = "ingress"
+  from_port                = 3128
+  to_port                  = 3128
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.reverse_proxy_ecs[0].id
+  security_group_id        = aws_security_group.internet_proxy_endpoint.id
+}
+
+resource "aws_security_group_rule" "reverse_proxy_http_ingress" {
+  count             = local.reverse_proxy_enabled[local.environment] ? 1 : 0
+  description       = "Reverse Proxy Container HTTP Rule"
+  type              = "ingress"
+  protocol          = "tcp"
+  from_port         = "80"
+  to_port           = "80"
+  cidr_blocks       = [module.vpc.vpc.cidr_block]
```

Comment:
Could we tighten this down? Isn't it just VPN or DirectConnect where the connections would originate from?

Reply:
No, it's a LB to ECS rule?

Reply:
Actually, if this doesn't have a LB SG now you may be right. ECS is new to me.